target/linux/ath79/files/drivers/gpio/gpio-rb91x-key.c

```diff
@@ -155,42 +153,23 @@ static int gpio_rb91x_key_probe(struct platform_device *pdev)
 	mutex_init(&drvdata->poll_mutex);
 
 	drvdata->gpio = devm_gpiod_get(dev, NULL, GPIOD_OUT_LOW);
-	if (IS_ERR(drvdata->gpio)) {
-		if (PTR_ERR(drvdata->gpio) != -EPROBE_DEFER) {
-			dev_err(dev, "failed to get gpio: %ld\n",
-				PTR_ERR(drvdata->gpio));
-		}
-		return PTR_ERR(drvdata->gpio);
-	}
+	if (IS_ERR(drvdata->gpio))
+		return dev_err_probe(dev, PTR_ERR(drvdata->gpio), "failed to get gpio");
```

Comment:
I guess we need to preserve `\n` here. Ref: https://elixir.bootlin.com/linux/v6.6.52/A/ident/dev_err_probe

Reply:
Hmm? I would think dev_err_probe appends the error to the string.